ocaml/ocamltest/actions_helpers.ml

```diff
@@ -17,19 +17,30 @@
 
 open Ocamltest_stdlib
 
+let fail_with_reason reason =
+  let code _log env =
+    let result = Result.fail_with_reason reason in
+    (result, env)
+  in
+  Actions.make ~name:"fail" ~description:"Make the test fail"
+    ~does_something:true
+    code
+
 let skip_with_reason reason =
   let code _log env =
     let result = Result.skip_with_reason reason in
     (result, env)
   in
-  Actions.make ~name:"skip" ~description:"Skip the test" code
+  Actions.make ~name:"skip" ~description:"Skip the test"
+    ~does_something:true
+    code
 
-let pass_or_skip test pass_reason skip_reason _log env =
+let predicate test pass_reason _skip_reason _log env =
   let open Result in
   let result =
     if test
-    then pass_with_reason pass_reason
-    else skip_with_reason skip_reason in
+    then predicate_satisfied_with_reason pass_reason
+    else predicate_not_satisfied_with_reason pass_reason in
```

Comment:
```suggestion
    else predicate_not_satisfied_with_reason skip_reason in
```

Reply:
(not sure; if not the parameter should
probably be removed)

Reply:
Actually we always want the pass-reason here, otherwise you get a double negation and the message printed is wrong.  I'll add a comment, but won't remove the parameter yet to minimize the diff with upstream.